### DISCUSSION TITLE
Support for keepLastN in DO_WHILE Task

### DISCUSSION THREAD

**Discussion by sriharimalagi** [conductor-oss/conductor#134] [Q&A]
Hello Team,

Problem statement:
- We are running DO_WHILE tasks for large number of iterations (>1000). When each DO_WHILE task iteration is completed, the output is stored into the [Task.outputData](https://github.com/conductor-oss/conductor/blob/main/core/src/main/java/com/netflix/conductor/model/TaskModel.java#L162).
- Since the results after each iteration is appended, the [Task Output Max Threshold limit](https://github.com/conductor-oss/conductor/blob/main/core/src/main/java/com/netflix/conductor/core/utils/ExternalPayloadStorageUtils.java#L119) is breached, causing the workflow to get terminated. 
- We would like to apply a solution such that previously executed iterations' output can be discarded.

Analysis:
- Upon reviewing the [DO_WHILE task documentation](https://orkes.io/content/reference-docs/operators/do-while#:~:text=The%20Do%20While%20task%20sequentially,while%20task%20in%20programming%20languages.), I have identified a property named keepLastN which is currently not supported in the open source code.
- Leveraging this logic for our DO_WHILE task would be beneficial. I am prepared to initiate a pull request for its implementation.

I kindly request your feedback on this matter.

Thank you and best regards,  
Srihari

**Comment by hchatillon** [comment]
Hello, did you find a solution for your issue ? I am interested in your approach

**Comment by hchatillon** [comment]
I was able to make it work, keepLastN has to be inserted in the inputParameters of the DO_WHILE task, and to keep nothing, you set it to -2 (don't ask why, the documentation is not that clear and the default value to 2)
